### PR TITLE
base.py: Add intel gpu information helpers

### PR DESCRIPTION
**Pull request by haoxli**

### CODE REVIEW

File: base.py [hunk 1]
```diff
@@ -1068,6 +1068,121 @@ def get_gpu_info():
...
+        # Gen 12
+        if masked_id == '0x9a00':
+            return 'tigerlake'
+        elif masked_id == '0x4c00':
+            return 'rocketlake'
+        elif masked_id == '0x4900':
+            return 'dg1'
+        elif masked_id == '0x4600':
+            return 'alderlake'
+        elif masked_id in {'0x4f00', '0x5600'}:
```
Comment (gyagp) [inline base.py:1084]: {} -> []?

**Comment by haoxli** [comment]
I copy all of gpu series types and generations from https://source.chromium.org/chromium/chromium/src/+/main:gpu/config/gpu_util.cc, but sort them by generations in descending order to get early return on latest devices. PTAL again!

File: base.py (hunk 1 above)
Comment (haoxli) [reply base.py:1084]: Use sets (`{}`) is better since the order doesn't matter here.

Comment (gyagp) [reply base.py:1084]: ok

**Pull request merged**